pkg/sdk/tags_dto_builders.go

```diff
@@ -0,0 +1,177 @@
+	s := AlterTagRequest{}
+	s.name = name
+	return &s
+}
+
+func (s *AlterTagRequest) WithAdd(values []string) *AlterTagRequest {
+	if len(values) > 0 {
+		s.add = &TagAdd{createAllowedValues(values)}
+	}
+	return s
+}
+
+func (s *AlterTagRequest) WithDrop(values []string) *AlterTagRequest {
+	if len(values) > 0 {
+		s.drop = &TagDrop{createAllowedValues(values)}
+	}
+	return s
+}
+
+func NewTagSetRequest() *TagSetRequest {
+	return &TagSetRequest{}
+}
+
+func (s *TagSetRequest) WithMaskingPolicies(maskingPolicies []string) *TagSetRequest {
+	s.maskingPolicies = maskingPolicies
+	return s
+}
+
+func (s *TagSetRequest) WithForce(force bool) *TagSetRequest {
+	s.force = Bool(force)
+	return s
+}
+
+func (s *TagSetRequest) WithComment(comment string) *TagSetRequest {
+	s.comment = String(comment)
+	return s
+}
+
+func createTagMaskingPolicies(policies []string) []TagMaskingPolicy {
+	items := make([]TagMaskingPolicy, 0, len(policies))
+	for _, policy := range policies {
+		items = append(items, TagMaskingPolicy{
+			Name: policy,
+		})
+	}
+	return items
+}
+
+func (s *AlterTagRequest) WithSet(request *TagSetRequest) *AlterTagRequest {
+	set := &TagSet{
+		Comment: request.comment,
+	}
+	if len(request.maskingPolicies) > 0 {
+		set.MaskingPolicies = &TagSetMaskingPolicies{
+			MaskingPolicies: createTagMaskingPolicies(request.maskingPolicies),
+			Force:           request.force,
+		}
+	}
+	s.set = set
+	return s
+}
+
+func NewTagUnsetRequest() *TagUnsetRequest {
+	return &TagUnsetRequest{}
+}
+
+func (s *TagUnsetRequest) WithMaskingPolicies(maskingPolicies []string) *TagUnsetRequest {
+	s.maskingPolicies = maskingPolicies
+	return s
+}
+
+func (s *TagUnsetRequest) WithAllowedValues(allowedValues bool) *TagUnsetRequest {
+	s.allowedValues = Bool(allowedValues)
+	return s
+}
+
+func (s *TagUnsetRequest) WithComment(comment bool) *TagUnsetRequest {
+	s.comment = Bool(comment)
+	return s
+}
+
+func (s *AlterTagRequest) WithUnset(request *TagUnsetRequest) *AlterTagRequest {
+	unset := &TagUnset{
+		AllowedValues: request.allowedValues,
+		Comment:       request.comment,
+	}
+	if len(request.maskingPolicies) > 0 {
+		unset.MaskingPolicies = &TagUnsetMaskingPolicies{
+			MaskingPolicies: createTagMaskingPolicies(request.maskingPolicies),
+		}
+	}
+	s.unset = unset
+	return s
+}
+
+func (s *AlterTagRequest) WithRename(name SchemaObjectIdentifier) *AlterTagRequest {
+	s.rename = &TagRename{
+		Name: name,
+	}
+	return s
+}
+
+func NewShowTagRequest() *ShowTagRequest {
+	return &ShowTagRequest{}
+}
+
+func (s *ShowTagRequest) WithLike(pattern string) *ShowTagRequest {
+	s.like = &Like{
+		Pattern: String(pattern),
+	}
+	return s
+}
+
+func (s *ShowTagRequest) WithIn(in *In) *ShowTagRequest {
+	s.in = in
+	return s
+}
+
+func NewDropTagRequest(name SchemaObjectIdentifier) *DropTagRequest {
+	s := DropTagRequest{}
+	s.name = name
+	return &s
+}
+
+func (s *DropTagRequest) WithIfNotExists(ifNotExists bool) *DropTagRequest {
```

Comment:
`WithIfExists`

Reply:
why? 
`CREATE [ OR REPLACE ] TAG [ IF NOT EXISTS ]`

its literally called "If not exists". Should we not be consistent with the naming convention?

Reply:
This is drop, not create. In drop, you have `DROP TAG [ IF EXISTS ] <name>`. The implementation underneath is good, but here, I assume it was a copy-paste error.

Reply:
oh yeah, you are right. fixed